wetterdienst/provider/dwd/derived/api.py

```diff
@@ -0,0 +1,327 @@
+# Copyright (C) 2018-2025, earthobservations developers.
+# Distributed under the MIT License. See LICENSE for more info.
+"""DWD derived data provider."""
+
+from __future__ import annotations
+
+import datetime as dt
+import logging
+from dataclasses import dataclass
+from datetime import datetime, UTC
+
+from itertools import groupby
+from typing import TYPE_CHECKING, ClassVar
+from zoneinfo import ZoneInfo
+
+import polars as pl
+import re
+
+from wetterdienst import Period
+from wetterdienst.metadata.cache import CacheExpiry
+from wetterdienst.metadata.resolution import Resolution
+from wetterdienst.model.request import TimeseriesRequest
+from wetterdienst.model.values import TimeseriesValues
+from wetterdienst.provider.dwd.derived.metadata import DwdDerivedMetadata
+from wetterdienst.provider.dwd.observation.metaindex import _read_meta_df
+from wetterdienst.util.enumeration import parse_enumeration_from_template
+from wetterdienst.util.network import download_file, list_remote_files_fsspec
+from wetterdienst.util.python import to_list
+
+if TYPE_CHECKING:
+    from wetterdienst.model.metadata import ParameterModel, DatasetModel
+
+log = logging.getLogger(__name__)
+
+_ENDPOINT_PREFIX = {
+    "heating_degreedays": {
+        Period.RECENT: "https://opendata.dwd.de/climate_environment/CDC/derived_germany/"
+        "techn/monthly/heating_degreedays/hdd_3807/recent/",
+        Period.HISTORICAL: "https://opendata.dwd.de/climate_environment/CDC/derived_germany/"
+        "techn/monthly/heating_degreedays/hdd_3807/historical/",
+    },
+    "stations_list": "https://opendata.dwd.de/climate_environment/CDC/help/KL_Monatswerte_Beschreibung_Stationen.txt",
+}
+
+_ENDPOINT_SUFFIX = {
+    "heating_degreedays": "gradtage_{date}.csv",
+}
+
+
+def _get_data_from_file(file):
+    df = pl.read_csv(
+        file.content,
+        separator=";",
+        skip_rows=3,
+    )
+    df.columns = pl.Series(df.columns).str.strip_chars()
+    return df
+
+
+class DwdDerivedValues(TimeseriesValues):
+    """Values class for dwd derived data."""
+
+    _default_start_dates: ClassVar = {
+        Resolution.MONTHLY: dt.datetime(2000, 1, 1, tzinfo=ZoneInfo("UTC")),
+    }
+
+    _column_name_mapping = {
+        "#ID": "station_id",
+        "Anzahl Tage": "amount_days_per_month",
+        "Monatsgradtage": "heating_degreedays",
+        "Anzahl Heiztage": "amount_heating_degreedays_per_month",
+    }
+
+    @staticmethod
+    def _get_values_for_single_station(
+        df: pl.DataFrame,
+        station_id: str,
+    ) -> pl.DataFrame | None:
+        """
+        Filter stations for a particular station.
+        """
+        try:
+            row_data = df.row(by_predicate=(pl.col("#ID") == int(station_id)), named=True)
+            return pl.DataFrame(row_data)
+        except pl.exceptions.NoRowsReturnedError:
+            return None
```

Comment:
Not sure if this function is needed, just use `.filter` method like

```suggestion
    df = df.filter(pl.col("#ID") == int(station_id))
    # or using `.eq` which I personally prefer 
    df = df.filter(pl.col("#ID").eq(int(station_id)))
```

Reply:
Done, thanks for the tip!